src/Store/Factory.php

```diff
@@ -51,12 +62,14 @@ public function getEntityManager() {
 		$timestampableListener->setAnnotationReader( $annotationReader );
 		$eventManager->addEventSubscriber( $timestampableListener );
 
+		$eventManager->addEventListener( 'prePersist', new TransferCodeListener( new TransferCodeGenerator() ) );
+
 		$entityManager = EntityManager::create( $this->connection, $config, $eventManager );
 
 		$platform = $entityManager->getConnection()->getDatabasePlatform();
 		$platform->registerDoctrineTypeMapping( 'enum', 'string' );
 
-		return $entityManager;
+		$this->entityManager = $entityManager;
```

Comment:
I'd return the created EM and have the caller set it to a field, as this is also where the field is used